spec/features/avo/belongs_to_spec.rb

```diff
@@ -159,6 +159,16 @@
     end
   end
 
+  describe "with custom primary key set" do
+    let(:event) { create(:event, name: "Sample Event") }
+
+    let!(:volunteer) { create(:volunteer, event: event) }
+
+    it "displays the event link using the UUID and stores the correct foreign key" do
+      expect(volunteer.event_uuid).to eq(event.uuid)
+    end
+  end
```

Comment:
This test is not effectively validating the changes. If we use the same setup on `main` without the logic changes to the `belongs_to` field, it still runs successfully without any failures.

Let's correctly set the volunteer with `field :event, as: :belongs_to` and write a test that ensures a volunteer is created, during creation it selects an event, and is correctly associated with that event after creation.